test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/ValidationSummaryTagHelperTest.cs

```diff
@@ -211,11 +325,18 @@ public static TheoryData<ModelStateDictionary> ProcessAsync_GeneratesExpectedOut
             await validationSummaryTagHelper.ProcessAsync(tagHelperContext, output);
 
             // Assert
-            Assert.Equal(2, output.Attributes.Count);
-            var attribute = Assert.Single(output.Attributes, attr => attr.Name.Equals("class"));
-            Assert.Equal("form-control validation-summary-errors", attribute.Value);
-            attribute = Assert.Single(output.Attributes, attr => attr.Name.Equals("data-valmsg-summary"));
-            Assert.Equal("true", attribute.Value);
+            Assert.Collection(
+                output.Attributes,
+                attribute =>
+                {
+                    Assert.Equal("class", attribute.Name);
+                    Assert.Equal("form-control validation-summary-errors", attribute.Value);
+                },
+                attribute =>
+                {
+                    Assert.Equal("data-valmsg-summary", attribute.Name);
+                    Assert.Equal("true", attribute.Value);
+                });
```

Comment:
Should always check order within `TagHelperOutput.Attributes`. `Assert.Single()` approach is harder to read anyhow.